src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceTestTransformer.java

```diff
@@ -174,4 +182,87 @@ private static ModelTypeTable createTypeTable(String packageName) {
     }
     return testCaseViews.build();
   }
+
+  private List<ViewModel> createSmokeTestViews(Model model, GapicProductConfig productConfig) {
+    ImmutableList.Builder<ViewModel> models = ImmutableList.builder();
+    SurfaceNamer surfacePackageNamer = new PythonSurfaceNamer(productConfig.getPackageName());
+    SurfaceNamer testPackageNamer =
+        new PythonSurfaceNamer(surfacePackageNamer.getTestPackageName());
+    for (Interface apiInterface : new InterfaceView().getElementIterable(model)) {
+      ModelTypeTable typeTable = createTypeTable(surfacePackageNamer.getTestPackageName());
+      GapicInterfaceContext context =
+          GapicInterfaceContext.create(
+              apiInterface, productConfig, typeTable, surfacePackageNamer, featureConfig);
+      if (context.getInterfaceConfig().getSmokeTestConfig() != null) {
+        models.add(createSmokeTestClassView(context, testPackageNamer));
+      }
+    }
+    return models.build();
+  }
+
+  private SmokeTestClassView createSmokeTestClassView(
+      GapicInterfaceContext context, SurfaceNamer testPackageNamer) {
+    String outputPath =
+        pathMapper.getOutputPath(context.getInterface(), context.getProductConfig());
+    SurfaceNamer namer = context.getNamer();
+    String name = namer.getSmokeTestClassName(context.getInterfaceConfig());
+
+    Method method = context.getInterfaceConfig().getSmokeTestConfig().getMethod();
+    FlatteningConfig flatteningGroup =
```

Comment:
This is more of a general question --- Why do we assume there is a flatteningGroup here? Does it mean that all smoke test method must flattened?